android/src/main/kotlin/io/eddayy/flutter_star_prnt/FlutterStarPrntPlugin.kt

```diff
@@ -306,128 +330,248 @@ public class FlutterStarPrntPlugin : FlutterPlugin, MethodCallHandler {
+        val diffusion: Boolean =
+            if (it.containsKey("diffusion")) (it.get("diffusion").toString()).toBoolean() else true
         val width: Int = if (it.containsKey("width")) (it.get("width").toString()).toInt() else 576
-        val bothScale: Boolean = if (it.containsKey("bothScale")) (it.get("bothScale").toString()).toBoolean() else true
-        val rotation: ICommandBuilder.BitmapConverterRotation = if (it.containsKey("rotation")) getConverterRotation(it.get("rotation").toString()) else getConverterRotation("Normal")
+        val bothScale: Boolean =
+            if (it.containsKey("bothScale")) (it.get("bothScale").toString()).toBoolean() else true
+        val rotation: ICommandBuilder.BitmapConverterRotation =
+            if (it.containsKey("rotation")) getConverterRotation(it.get("rotation").toString())
+            else getConverterRotation("Normal")
         try {
-            var bitmap: Bitmap? = null
-            if (URLUtil.isValidUrl(it.get("appendBitmap").toString())) {
-              val imageUri: Uri = Uri.parse(it.get("appendBitmap").toString())
-              bitmap  = MediaStore.Images.Media.getBitmap(context.getContentResolver(), imageUri)
-            } else {
-              bitmap = BitmapFactory.decodeFile(it.get("appendBitmap").toString())
-            }
-
-            if (bitmap != null) {
-              if (it.containsKey("absolutePosition")) {
-                builder.appendBitmapWithAbsolutePosition(bitmap, diffusion, width, bothScale, rotation, (it.get("absolutePosition").toString()).toInt())
-              } else if (it.containsKey("alignment")) {
-                  builder.appendBitmapWithAlignment(bitmap, diffusion, width, bothScale, rotation, getAlignment(it.get("alignment").toString()))
-              } else builder.appendBitmap(bitmap, diffusion, width, bothScale, rotation)
-            }
+            val imageUri: Uri = Uri.parse(it.get("appendBitmap").toString())
+            val bitmap: Bitmap = MediaStore.Images.Media.getBitmap(context.getContentResolver(), imageUri)
+            if (it.containsKey("absolutePosition")) {
+              builder.appendBitmapWithAbsolutePosition(
+                  bitmap,
+                  diffusion,
+                  width,
+                  bothScale,
+                  rotation,
+                  (it.get("absolutePosition").toString()).toInt())
+            } else if (it.containsKey("alignment")) {
+              builder.appendBitmapWithAlignment(
+                  bitmap,
+                  diffusion,
+                  width,
+                  bothScale,
+                  rotation,
+                  getAlignment(it.get("alignment").toString()))
+            } else builder.appendBitmap(bitmap, diffusion, width, bothScale, rotation)
         } catch (e: Exception) {
           Log.e("FlutterStarPrnt", "appendbitmap failed", e)
         }
+      } else if (it.containsKey("appendBitmapImg")) {
```

Comment:
the android code also has `appendBitmapByteArray` doing the same thing

Reply:
Removed and committed